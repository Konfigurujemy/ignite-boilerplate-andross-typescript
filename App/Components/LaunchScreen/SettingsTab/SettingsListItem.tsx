import * as React from "react";
import { Row, Text, CardItem} from 'native-base';
import Fonts from "../../../Themes/Fonts";
import MaterialIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {ColorScheme} from "../../../Themes/Colors";


interface SettingsListItemProps {
  name: string,
  icon: string,
  // onPress(): void,
  colorScheme:ColorScheme
}

export default (item: SettingsListItemProps) => {
  return (
    <CardItem button bordered>
      <Row style={{alignItems:'center',justifyContent:'center'}}>


        <MaterialIcons
          name={item.icon}
          size={28}
          style={{margin: 8}}
          color={'#424242'}
        />
        <Text style={{fontFamily: Fonts.type.farsi, color: '#424242',flex:1}}>{item.name}</Text>
      </Row>

    </CardItem>
  )
    ;
}
