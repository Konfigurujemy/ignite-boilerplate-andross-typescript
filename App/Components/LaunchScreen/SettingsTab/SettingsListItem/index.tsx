import Ltr from "./SettingsListItem.ltr";
import Rtl from "./SettingsListItem.rtl";
import {connect} from "react-redux";
import * as React from "react";
import {ColorScheme} from "../../../../Themes/Colors";

export interface SettingsListItemProps {
  name: string,
  icon: string,
  colorScheme:ColorScheme,
  isRtl:boolean
}

export const Direction = (props:SettingsListItemProps) => props.isRtl? <Rtl {...props} /> : <Ltr {...props}/>;

const mapStateToProps = state => {
  return {
    isRtl: state.appSettings.isRtl
  };
};

export default connect(mapStateToProps)(Direction);
