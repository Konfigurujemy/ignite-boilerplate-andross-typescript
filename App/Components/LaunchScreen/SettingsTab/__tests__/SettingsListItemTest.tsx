import "react-native";
import * as React from "react";
import {Direction as SettingsListItem} from "../SettingsListItem";
import * as renderer from 'react-test-renderer';
import {colorScheme} from "../../../../Themes/Colors";

test("Renders Correctly", () => {
  const tree = renderer.create(<SettingsListItem
      name={'sajjad'}
      icon={'1'}
      colorScheme={colorScheme(false)}
      isLtr={true}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
