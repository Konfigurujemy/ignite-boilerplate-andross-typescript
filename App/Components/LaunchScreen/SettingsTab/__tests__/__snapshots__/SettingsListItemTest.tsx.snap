// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderRadius": 2,
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#424242",
            "fontSize": 28,
          },
          Object {
            "margin": 8,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      e
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#424242",
            "flex": 1,
            "fontFamily": "BYekan+",
          },
        ]
      }
      uppercase={false}
    >
      sajjad
    </Text>
  </View>
</View>
`;
